pkg/resource/cluster/hooks.go

```diff
@@ -160,118 +160,177 @@ func (rm *resourceManager) customUpdate(
-			TargetBrokerEBSVolumeInfo: []svcsdktypes.BrokerEBSVolumeInfo{
-				{
-					KafkaBrokerNodeId: aws.String("ALL"),
-					VolumeSizeGB:      aws.Int32(int32(*desired.ko.Spec.BrokerNodeGroupInfo.StorageInfo.EBSStorageInfo.VolumeSize)),
-				},
-			},
-		})
-		rm.metrics.RecordAPICall("UPDATE", "UpdateBrokerStorage", err)
-		if err != nil {
-			return nil, err
-		}
-		message := fmt.Sprintf("kafka is updating broker storage")
-		ackcondition.SetSynced(updatedRes, corev1.ConditionFalse, &message, nil)
-		err = requeueAfterAsyncUpdate()
+		return rm.updateBrokerStorage(ctx, updatedRes, latest)
 
 	case delta.DifferentAt("Spec.BrokerNodeGroupInfo.InstanceType"):
-		_, err := rm.sdkapi.UpdateBrokerType(ctx, &svcsdk.UpdateBrokerTypeInput{
-			ClusterArn:         (*string)(latest.ko.Status.ACKResourceMetadata.ARN),
-			CurrentVersion:     latest.ko.Status.CurrentVersion,
-			TargetInstanceType: desired.ko.Spec.BrokerNodeGroupInfo.InstanceType,
-		})
-		rm.metrics.RecordAPICall("UPDATE", "UpdateBrokerType", err)
-		if err != nil {
-			return nil, err
-		}
-		message := fmt.Sprintf("kafka is updating broker instanceType")
-		ackcondition.SetSynced(updatedRes, corev1.ConditionFalse, &message, nil)
-		err = requeueAfterAsyncUpdate()
+		return rm.updateBrokerType(ctx, desired, latest)
 
 	case delta.DifferentAt("Spec.NumberOfBrokerNodes"):
-		_, err := rm.sdkapi.UpdateBrokerCount(ctx, &svcsdk.UpdateBrokerCountInput{
-			ClusterArn:                (*string)(latest.ko.Status.ACKResourceMetadata.ARN),
-			CurrentVersion:            latest.ko.Status.CurrentVersion,
-			TargetNumberOfBrokerNodes: aws.Int32(int32(*desired.ko.Spec.NumberOfBrokerNodes)),
-		})
-		rm.metrics.RecordAPICall("UPDATE", "UpdateBrokerCount", err)
-		if err != nil {
-			return nil, err
-		}
-		message := fmt.Sprintf("kafka is updating broker instanceType")
-		ackcondition.SetSynced(updatedRes, corev1.ConditionFalse, &message, nil)
-		err = requeueAfterAsyncUpdate()
+		return rm.updateNumberOfBrokerNodes(ctx, desired, latest)
+	}
+
+	return updatedRes, nil
+}
+
+// updateNumberOfBrokerNodes updates the number of broker
+// nodes for the kafka cluster
+func (rm *resourceManager) updateNumberOfBrokerNodes(
+	ctx context.Context,
+	desired *resource,
+	latest *resource,
+) (updatedRes *resource, err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.updateNumberOfBrokerNodes")
+	defer func() { exit(err) }()
 
+	_, err = rm.sdkapi.UpdateBrokerCount(ctx, &svcsdk.UpdateBrokerCountInput{
+		ClusterArn:                (*string)(latest.ko.Status.ACKResourceMetadata.ARN),
+		CurrentVersion:            latest.ko.Status.CurrentVersion,
+		TargetNumberOfBrokerNodes: aws.Int32(int32(*desired.ko.Spec.NumberOfBrokerNodes)),
+	})
+	rm.metrics.RecordAPICall("UPDATE", "UpdateBrokerCount", err)
+	if err != nil {
+		return latest, err
 	}
+	message := fmt.Sprintf("kafka is updating broker instanceType")
+	ackcondition.SetSynced(updatedRes, corev1.ConditionFalse, &message, nil)
```

Comment:
can we pull the "setCondtion' logic outside of the function to make it consistent with the rest?

Reply:
both approaches work, it's just neat to have consistency :)